_posts/2019-09-26-2019-building-admissions.md

```diff
@@ -0,0 +1,284 @@
+## Building Elixir School's Admissions application
+
+If you didn't know, Elixir School has its own Slack where contributors can gather to discuss our organization's content and projects but most importantly, support one another in our Elixir journey. When we set out to create our own Slack we wanted to address a big concern with many public Slacks: the signal to noise ratio is bad, there's just too much spam. 
+
+> Have you contributed to an Elixir School project but not joined us on Slack? Head over to https://admissions.elixirschool.com to get your invite today!
+
+So how can we keep our Slack public but prevent spammers from joining and do so in a way that doesn't add work to our maintainers? Our solution: required at least once contribution of any side to any one of our projects. 
+
+Achieving this required an application that used GitHub to verify a user's eligibility. This application would come to be known as: Admissions.
+
+> Want to skip ahead and see the final product? The code can be found at https://github.com/elixirschool/admissions.
+
+In this post we're going to explore how Admissions works and how we achieved our goals using Elixir and Phoenix. To start let's look at the expected flow and work from there:
+
+```mermaid
+graph LR
+    A["Sign in with GitHub"] --> B{"Contributor?"}
+    B-- Yes -->C["Send invite via Slack API"]
+    B-- No -->D["Ineligibility message"]
+    C-->E["Confirm email address"]
+    E-->F["Welcome message"]
+```
+
+In addition to telling us how the application should function, this diagram breaks the flow up into convenient development task. Working from this diagram let's explore the individual tasks we'll need in order to fulfill our high level requirements:
+
+1. Allow a user to sign in using GitHub and capture their access token. We can leverage [Ueberauth](https://github.com/ueberauth/ueberauth) and its [GitHub strategy](https://github.com/ueberauth/ueberauth_github) to do the heavy lifting for us.
+2. With the user's access token use the GitHub API to see if the user has contributed to an organization's project. To avoid having to spend time writing our own GitHub API client we're going to make use of [Tentacat](https://github.com/edgurgel/tentacat).
+3. Using the result of the API search, process the user's result
+   1. In the event a user **is** a contributor, have them confirm the email address they want to use for Slack, use the Slack API to send an invite, and finally congratulate them.
+   2. If they **have not** contributed we need to notify them of their ineligibility
+
+> Note: For our application we've renamed `PageController` to `EligibilityController` to fit with our school admissions theme.
```

Comment:
Not sure if you need this note. I don't think its confusing that you've built an `EligibilityController`/no expectation that a PageController should be handling auth.